app/models/user.rb

```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  validates :okta_id, uniqueness: true, presence: true
+
+  # while the email needs to be validated case-insensitively, we'll
+  # let Rails pass the insensitive check down to postgres's citext type
+  validates :email, uniqueness: {case_sensitive: true}, presence: true
```

Comment:
by default rails makes an inificent uniqueness check if you use `case_sensitive: false` (`where('LOWER(email) = ?', email.downcase)`), but if you use `case_sensitive: true` it uses the slightly faster `where('email = ?', email)`. This ends up being equivalent behavior on the database because the field is treated as case insensitive. does that make sense? (answering questions like this help me get better at communicating about it 😀)

Reply:
Yes, you explained this well. I understand now. Calling true on case_sensitive helps the sql query be more efficient, but the database still accomplishes the tasks of comparing insensitive case emails.